internal/interfaces/protocols_s3_policy.go

```diff
@@ -0,0 +1,269 @@
+type ProtocolsS3PolicyGetDataModelONTAP struct {
+    Name       string                              `mapstructure:"name"`
+    Comment    string                              `mapstructure:"comment,omitempty"`
+    Statements []ProtocolsS3PolicyStatementONTAP   `mapstructure:"statements,omitempty"`
+    SVM        ProtocolsS3PolicySVMONTAP           `mapstructure:"svm"`
+}
+
+// ProtocolsS3PolicyStatementONTAP describes policy statement structure
+type ProtocolsS3PolicyStatementONTAP struct {
+    Effect      string                     `mapstructure:"effect"`
+    Actions     []string                   `mapstructure:"actions,omitempty"`
+    Resources   []string                   `mapstructure:"resources,omitempty"`
+    SID         string                     `mapstructure:"sid,omitempty"`
+}
+
+// ProtocolsS3PolicySVMONTAP describes the SVM data model
+type ProtocolsS3PolicySVMONTAP struct {
+    Name string `mapstructure:"name"`
+    UUID string `mapstructure:"uuid,omitempty"`
+}
+
+// ProtocolsS3PolicyResourceBodyDataModelONTAP describes the body data model using go types for mapping.
+type ProtocolsS3PolicyResourceBodyDataModelONTAP struct {
+    Name       string                              `mapstructure:"name"`
+    Comment    string                              `mapstructure:"comment,omitempty"`
+    Statements []ProtocolsS3PolicyStatementONTAP   `mapstructure:"statements,omitempty"`
+}
+
+// ProtocolsS3PolicyDataSourceModel describes the data source data model.
+type ProtocolsS3PolicyDataSourceModel struct {
+    CxProfileName types.String                          `tfsdk:"cx_profile_name"`
+    Name          types.String                          `tfsdk:"name"`
+    Comment       types.String                          `tfsdk:"comment"`
+    SVMName       types.String                          `tfsdk:"svm_name"`
+    Statements    []ProtocolsS3PolicyStatementModel     `tfsdk:"statements"`
+    SVM           ProtocolsS3PolicySVMModel             `tfsdk:"svm"`
+    ID            types.String                          `tfsdk:"id"`
+}
+
+// ProtocolsS3PolicyStatementModel describes the statement model
+type ProtocolsS3PolicyStatementModel struct {
+    Effect      types.String              `tfsdk:"effect"`
+    Actions     []types.String            `tfsdk:"actions"`
+    Resources   []types.String            `tfsdk:"resources"`
+    SID         types.String              `tfsdk:"sid"`
+}
+
+// ProtocolsS3PolicySVMModel describes the SVM model
+type ProtocolsS3PolicySVMModel struct {
+    Name types.String `tfsdk:"name"`
+    UUID types.String `tfsdk:"uuid"`
+}
+
+type ProtocolsS3PolicyDataSourceFilterModel struct {
+    Name    types.String `tfsdk:"name"`
+    SVMName types.String `tfsdk:"svm_name"`
+}
+
+// getSVMUUID is a helper function to get SVM UUID from SVM name
+func getSVMUUID(errorHandler *utils.ErrorHandler, r restclient.RestClient, svmName string) (string, error) {
```

Comment:
This function is duplicated with our existing function. You may use [GetSvmByName](https://github.com/NetApp/terraform-provider-netapp-ontap/blob/integration/main/internal/interfaces/svm.go#L97)